src/features/click-to-play.js

```diff
@@ -2337,16 +2324,28 @@ async function createYouTubePreview (originalElement, widget) {
     return { youTubePreview, shadowRoot }
 }
 
+// Convention is that each function should be named the same as the sendMessage
+// method we are calling into eg. calling `sendMessage('getClickToLoadClicks')`
+// will result in a response routed to `updateHandlers.getClickToLoadClicks()`.
 const updateHandlers = {
-    // Convention is that each function should be named the same as the sendMessage method we are calling into
-    // eg. calling sendMessage('initClickToLoad') will result in a response routed to 'updateHandlers.initClickToLoad()'
-    initClickToLoad: function (resp) {
+    getClickToLoadClicks (response) {
+        for (const [entity, clickCount] of Object.entries(response)) {
+            if (entityData[entity]) {
+                entityData[entity].simpleVersion =
+                    clickCount >= entityData[entity].maxClicks
+            }
+        }
+
+        // TODO: Move the below init logic, it's confusing having it inside the
```

Comment:
+1 :)

Or call this something generic like "getCTLSettings" and have it return a hash of all the things.

Reply:
Good idea, I've combined them into `getClickToLoadState` which fixes the race condition + further reduces the messages required on init :+1:.